cli_tools/gce_vm_image_import/importer/inflater.go

```diff
@@ -112,16 +113,27 @@ func createDaisyInflater(args ImportArguments) (inflater, error) {
 		addFeatureToDisk(wf, "WINDOWS", inflationDiskIndex)
 	}
 
-	return daisyInflater{
+	return &daisyInflater{
 		wf:              wf,
 		inflatedDiskURI: fmt.Sprintf("zones/%s/disks/%s", args.Zone, diskName),
 	}, nil
 }
 
-func (inflater daisyInflater) traceLogs() []string {
+func (inflater *daisyInflater) traceLogs() []string {
 	return inflater.serialLogs
 }
 
+func (inflater *daisyInflater) cancel(reason string) {
+	inflater.wf.IterateWorkflowSteps(func(step *daisy.Step) {
+		if step.CreateDisks != nil {
+			for _, disk := range *step.CreateDisks {
+				disk.NoCleanup = false
+			}
+		}
+	})
+	inflater.wf.CancelWithReason(reason)
```

Comment:
NoCleanup is set to true here because the inflated disk is needed for the processing step (translate or just image create). If the workflow is cancelled, this disk should be cleaned up. Otherwise, temporary disks would keep accumulating and customers would keep incurring charges. The only reason to keep temporary disks is debugging, which doesn't apply here.

Reply:
I'm glad we're removing the disk; my concern is about perpetuating the pattern of modifying workflows after they've started.

Reply:
Your second question to Andrew was how would he feel about deleting all disks, even if `NoCleanup` is set to `true`. I've answered that question in my previous comment.

Regarding the mechanism of deletion, I agree using workflow modification is not the prettiest and goes against what we agreed on regarding Daisy usage. I'll change it to avoid workflow modification.